### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by mulimoen**
This fixes various CI failures, and adds a build for windows 1.12

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -124,6 +126,9 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v2
         with: {submodules: true}
+      - name: Change to older toolchain
+        if: matrix.os == 'macos'
+        run: sudo xcode-select -s "/Applications/Xcode_11.7.app"
```
Comment (aldanor) [inline .github/workflows/ci.yml:131]: @mulimoen Just wondering why is this needed?

**Review by aldanor — APPROVED** [review]
Looks great, thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mulimoen) [reply .github/workflows/ci.yml:131]: The error message: `implicit declaration of function 'H5O__fsinfo_set_version' is invalid in C99 `.

The updated compiler is a bit better at enforcing `c99` standards, and `hdf5` placed the function definition below first use. This should be fixed in updated version of `hdf5`, we should update `hdf5-src` to pull the official fix.

Comment (mulimoen) [reply .github/workflows/ci.yml:131]: See #130 for how to fix this permanently
